### ISSUE TITLE
ValueError: invalid literal for int() with base 10: '[Not Supported]'

### ISSUE THREAD

**Issue by weiweikong** [wookayin/gpustat#6]
Running `gpustat` got following return

```
Traceback (most recent call last):
  File "/home/~~~/anaconda2/bin/gpustat", line 11, in <module>
    sys.exit(main())
  File "/home/~~~/anaconda2/lib/python2.7/site-packages/gpustat.py", line 283, in main
    print_gpustat(**vars(args))
  File "/home/~~~/anaconda2/lib/python2.7/site-packages/gpustat.py", line 242, in print_gpustat
    gpu_stats.update_process_information()
  File "/home/~~~/anaconda2/lib/python2.7/site-packages/gpustat.py", line 196, in update_process_information
    processes = self.running_processes()
  File "/home/~~~/anaconda2/lib/python2.7/site-packages/gpustat.py", line 166, in running_processes
    pid_map = {int(e['pid']) : None for e in process_entries}
  File "/home/~~~/anaconda2/lib/python2.7/site-packages/gpustat.py", line 166, in <dictcomp>
    pid_map = {int(e['pid']) : None for e in process_entries}
ValueError: invalid literal for int() with base 10: '[Not Supported]'
```
- Ubuntu 14.04
- Python 2.7 with Anaconda 4

**Comment by R0flcopt3r** [comment]
Same issue here.

Nvidia driver: 361.42
Ubuntu 16.04

`> ./gpustat.py 
Traceback (most recent call last):
  File "./gpustat.py", line 317, in <module>
    main()
  File "./gpustat.py", line 314, in main
    print_gpustat(**vars(args))
  File "./gpustat.py", line 286, in print_gpustat
    gpu_stats = GPUStatCollection.new_query()
  File "./gpustat.py", line 153, in new_query
    return GPUStatCollection(gpu_list)
  File "./gpustat.py", line 126, in __init__
    self.update_process_information()
  File "./gpustat.py", line 203, in update_process_information
    processes = self.running_processes()
  File "./gpustat.py", line 173, in running_processes
    pid_map = {int(e['pid']) : None for e in process_entries}
  File "./gpustat.py", line 173, in <dictcomp>
    pid_map = {int(e['pid']) : None for e in process_entries}
ValueError: invalid literal for int() with base 10: '[Not Supported]'`

**Comment by R0flcopt3r** [comment]
Same issue here.

Ubuntu 16.04
nvidia 361.42

**Comment by wookayin** [comment]
@R0flcopt3r It looks that `nvidia-smi` returns `[Not Supported]` for process ID in some cases.
Could you please report the result of the following command when the problem occurs?

```
nvidia-smi --query-compute-apps=gpu_uuid,pid,used_memory --format=csv,noheader,nounits
```

**Comment by R0flcopt3r** [comment]
`> nvidia-smi --query-compute-apps=gpu_uuid,pid,used_memory --format=csv,noheader,nounits
GPU-97055bae-db68-136e-6729-49ac3c3fec08, [Not Supported], [Not Supported]`

`> nvidia-smi --query-gpu=index,uuid,name,temperature.gpu,utilization.gpu,memory.used,memory.total --format=csv,noheader,nounits
0, GPU-97055bae-db68-136e-6729-49ac3c3fec08, GeForce GTX 780, 55, [Not Supported], 339, 3068
`

**Comment by weiweikong** [comment]
@R0flcopt3r 
```
07:47:19 $ nvidia-smi --query-gpu=index,uuid,name,temperature.gpu,utilization.gpu,memory.used,memory.total --format=csv,noheader,nounits
0, GPU-f177935b-3adb-ea65-559a-1f25b780296e, Tesla K40c, 44, 0, 23, 11519
1, GPU-806fd873-9f2a-1ded-e880-1977ce85632d, GeForce GT 610, 41, [Not Supported], 446, 1016
2, GPU-5428f7cf-1463-8546-7cb3-58ad7332aaca, Tesla K40c, 47, 0, 23, 11519
```
```
07:48:26 $ nvidia-smi --query-compute-apps=gpu_uuid,pid,used_memory --format=csv,noheader,nounits
GPU-806fd873-9f2a-1ded-e880-1977ce85632d, [Not Supported], [Not Supported]
```
some `[Not Supported] appears. It seems that the `nvidia-smi` dose not return the value which `gpustat` needed.

**Comment by wookayin** [comment]
It looks that `nvidia-smi` cannot retrieve some usage information of older GPU cards such as GT 610 and GTX 780. I fixed the issue in 902f9cd. 

Please try with the latest master version, or you can install it directly via pip+git:
```
pip install git+https://github.com/wookayin/gpustat.git@master
```

**Comment by weiweikong** [comment]
@wookayin 
It works fine. :)
```
~  Mon Nov  7 17:21:40 2016
[0] Tesla K40c       | 44'C,   0 % |    23 / 11519 MB |
[1] GeForce GT 610   | 41'C,  ?? % |   397 /  1016 MB | --/--(?M)
[2] Tesla K40c       | 46'C,   0 % |    23 / 11519 MB |
```
